test/tic_tac_toe_test.exs

```diff
@@ -2,7 +2,32 @@ defmodule TicTacToeTest do
   use ExUnit.Case
   import ExUnit.CaptureIO
 
-  test "displays the welcome message" do
-    assert capture_io(fn -> TicTacToe.welcome_message() end) === "Welcome to Tic-Tac-Toe"
+
+  describe "start/1" do
+    test "displays welcome message and board" do
+      assert capture_io(fn -> TicTacToe.start(CommandLine) end) ===
+        """
+        Welcome to Tic-Tac-Toe
+
+        1 | 2 | 3
+        --|---|--
+        4 | 5 | 6
+        --|---|--
+        7 | 8 | 9
+        """
+    end
+
+    test "displays welcome message and board using mock" do
```

Comment:
I don't think we need the `using mock` part here. `TicTacToe.start` should work with any `writer` that conforms to the interface(DisplayBehaviour) so we should treat `CommandLineMock` as just another `writer`.

Reply:
I completely agree that the TicTacToe.start should work with any writer that conforms to the DisplayBehaviour interface, but isn't this test still going to use a mock of whatever writer the TicTacToe.start would take? I think in this case the `using mock` part could be useful to specify.

Reply:
Assuming the other test above is going to be moved to the `CommandLineTest` (so there won't be a duplicate test description in this file), then I think it makes sense to drop the "using mock" portion of the test description as well. We are using a mock here, but as you continue to leverage mocks in your tests this verbiage in the descriptions would start to become quite repetitive. Mocks are a tool we use to make unit testing possible, but they aren't really relevant for describing the behavior under test, which is what I would want to see in the test description.

Reply:
I like this! Descriptive and seems like a good convention to have in the repo moving forward!

Reply:
Ohhhh ok, this makes sense to me.